refaster-support/src/main/java/tech/picnic/errorprone/refaster/AnnotatedCompositeCodeTransformer.java

```diff
@@ -0,0 +1,153 @@
+package tech.picnic.errorprone.refaster;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+import static com.google.errorprone.BugPattern.SeverityLevel.SUGGESTION;
+import static com.google.errorprone.BugPattern.SeverityLevel.WARNING;
+import static tech.picnic.errorprone.refaster.annotation.OnlineDocumentation.NESTED_CLASS_URL_PLACEHOLDER;
+import static tech.picnic.errorprone.refaster.annotation.OnlineDocumentation.TOP_LEVEL_CLASS_URL_PLACEHOLDER;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Comparators;
+import com.google.common.collect.ImmutableClassToInstanceMap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterators;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.CodeTransformer;
+import com.google.errorprone.CompositeCodeTransformer;
+import com.google.errorprone.DescriptionListener;
+import com.google.errorprone.ErrorProneOptions;
+import com.google.errorprone.matchers.Description;
+import com.sun.source.util.TreePath;
+import com.sun.tools.javac.util.Context;
+import java.io.Serializable;
+import java.lang.annotation.Annotation;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.function.Function;
+import tech.picnic.errorprone.refaster.annotation.OnlineDocumentation;
+import tech.picnic.errorprone.refaster.annotation.Severity;
+
+/**
+ * A {@link CompositeCodeTransformer} that augments the {@link Description} of Refaster rule
+ * matches.
+ *
+ * <p>The content is augmented based on custom {@link tech.picnic.errorprone.refaster.annotation
+ * annotations} available on the matching {@link CodeTransformer} or on this {@link
+ * CompositeCodeTransformer} as a fallback, if any.
+ */
+// XXX: Test this class directly. (Right now it's only indirectly tested through `RefasterTest`.)
```

Comment:
Is this an XXX we wish to resolve as part of this PR? Haven't gave it much thought, but I feel the approach will be quite the same (have some dummy Refaster rules with these annotations).  I'm okay with leaving it like this (we clearly state the connection), or, if it makes sense, to move some of the introduced testcases from `RefasterTest` to a new test class.

Reply:
My thinking is that we can test this class using a "lower-level" integration, possibly mocks. I'm okay with deferring that work to another PR, though, so merging isn't blocked on this. (But who knows, perhaps I'll have a look at this later today.)

Reply:
I added a commit for this. A "smaller" `XXX` comment remains, but that's for another day.